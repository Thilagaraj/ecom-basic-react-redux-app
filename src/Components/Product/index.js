import React from 'react';
import {connect} from 'react-redux';
import { Card, CardImg, CardText, CardBody, CardTitle, CardSubtitle, Button} from 'reactstrap';
import {Link} from 'react-router-dom';
import {AddToCart} from '../../Actions/Actions';
class Product extends React.Component { 
	constructor(props){
		super(props);
		this.handleCartClick=this.handleCartClick.bind(this);
	}
	
	handleCartClick(event,product){
		//this.props.dispatch({type:'ADD_TO_CART',payload:product});	
		this.props.addToCart(product);
	}
	
	render(){
		let product=this.props.details;
		return (
			 <div>
			  <Card>
				<CardImg top width="100%" src={product.picture} alt="Card image cap" />
				<CardBody>
				  <CardTitle><Link to={`/details/${product.id}`}>{product.title}</Link></CardTitle>
				  <CardSubtitle>{product.category}</CardSubtitle>
				  <CardText>{product.description.slice(0,100)}</CardText>
				  <Button onClick={(e)=>this.handleCartClick(e,product)}>Add to cart</Button>
				</CardBody>
			  </Card>
		</div>
		);
	}
	
}
const mapDispatchToProps=(dispatch)=>{
	return {
		addToCart:(product)=>{dispatch(AddToCart(product))}
	}
}
export default connect(null,mapDispatchToProps)(Product);