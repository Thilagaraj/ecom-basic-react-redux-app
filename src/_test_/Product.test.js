import React from 'react';
import { shallow } from 'enzyme';
import Product from '../Components/Product';

test("component just rendering",()=>{
	const wrapper=shallow(<product/>);
	expect(wrapper).toMatchSnapshot();	
});
test("component rendering with product props",()=>{
	const detls={
    "id": "5b6eb27314c4ed27570b601e",
    "isActive": true,
    "price": "$2,752.35",
    "picture": "http://placehold.it/250x250",
    "title": "Enid Vazquez",
    "category": "apple",
    "description": "dolor eiusmod sint duis culpa non aute enim enim Lorem eu nisi non aliquip exercitation reprehenderit laboris ipsum qui enim nisi elit ullamco qui aliqua veniam adipisicing dolor labore excepteur enim aliqua veniam ipsum id enim do ex nisi proident",
    "tags": [
      "minim",
      "laboris",
      "ipsum",
      "non",
      "dolore",
      "ipsum",
      "duis"
    ]
};
	const wrapper=shallow(<product details={detls}/>);
	expect(wrapper).toMatchSnapshot();	
});