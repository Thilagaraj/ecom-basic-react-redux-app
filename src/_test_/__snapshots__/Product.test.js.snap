// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component just rendering 1`] = `<product />`;

exports[`component rendering with product props 1`] = `
<product
  details={
    Object {
      "category": "apple",
      "description": "dolor eiusmod sint duis culpa non aute enim enim Lorem eu nisi non aliquip exercitation reprehenderit laboris ipsum qui enim nisi elit ullamco qui aliqua veniam adipisicing dolor labore excepteur enim aliqua veniam ipsum id enim do ex nisi proident",
      "id": "5b6eb27314c4ed27570b601e",
      "isActive": true,
      "picture": "http://placehold.it/250x250",
      "price": "$2,752.35",
      "tags": Array [
        "minim",
        "laboris",
        "ipsum",
        "non",
        "dolore",
        "ipsum",
        "duis",
      ],
      "title": "Enid Vazquez",
    }
  }
/>
`;
